src/controllers/connectionManager.ts

```diff
@@ -289,6 +289,18 @@ export default class ConnectionManager {
 		return ConnectionCredentials.createConnectionDetails(connectionInfo);
 	}
 
+	/**
+	 * Send a request to the SQL Tools Server client
+	 * @param type The of the request to make
+	 * @param params The params to pass with the request
+	 * @returns A thenable object for when the request receives a response
+	 */
+	public sendRequest<P, R, E, R0>(requestType: RequestType<P, R, E, R0>, params?: P): Thenable<R> {
+		if (this.client !== undefined) {
```

Comment:
`this.client?.sendRequest(...)`

But also - why is this necessary? I wouldn't expect client to ever be undefined. It's passed in or set in the constructor.